src/Build.UnitTests/Graph/IsolateProjects_Tests.cs

```diff
@@ -239,7 +239,28 @@ public void IsolationRelatedMessagesShouldNotBePresentInNonIsolatedBuilds()
                     logger.AssertLogDoesntContain("MSB4260");
                 },
                 excludeReferencesFromConstraints: true,
-                isolateProjects: false);
+                isolateProjects: ProjectIsolationMode.False);
+        }
+
+        [Fact]
+        public void IsolationRelatedMessageShouldBePresentInIsolatedBuildsWithMessaging()
```

Comment:
I would like to see another new test in this file with the end-to-end scenario: an undeclared reference that alters the return value of the declared reference's target but that doesn't propagate to the caller because of the caching.